tests/test_users.py

```diff
@@ -0,0 +1,232 @@
+#! env/bin/python
+
+import json
+import unittest
+
+from tests.base_test import BaseTestCase
+
+
+class TestGetAllUsers(BaseTestCase):
+
+    def test_get_users_with_no_token(self):
+        response = self.client.get('/api/users')
+        self.assertEqual(response.status_code, 401)
+
+    def test_get_users_with_empty_token(self):
+        response = self.client.get('/api/users', headers={'token': ''})
+        self.assertEqual(response.status_code, 401)
+
+    def test_get_users_with_invalid_token(self):
+        token = self.login().json['token'] + '1'
+        response = self.client.get('/api/users', headers={'token': token})
+        self.assertEqual(response.status_code, 401)
+
+    def test_get_users_with_valid_token(self):
+        token = self.login().json['token']
+        response = self.client.get('/api/users', headers={'token': token})
+        self.assertEqual(response.status_code, 200)
+
+    def test_response_contains_users(self):
+        token = self.login().json['token']
+        response = self.client.get('/api/users', headers={'token': token})
+        self.assertEqual(json.loads(response.data)[0]["name"], "user1")
+
+
+class TestPostUser(BaseTestCase):
+
+    def test_post_user_with_no_token(self):
+        response = self.client.post('/api/users')
+        self.assertEqual(response.status_code, 401)
+
+    def test_gpost_users_with_empty_token(self):
+        response = self.client.post('/api/users', headers={'token': ''})
+        self.assertEqual(response.status_code, 401)
+
+    def test_post_users_with_invalid_token(self):
+        token = self.login().json['token'] + '1'
+        response = self.client.post(
+            '/api/users', headers={'token': token}, data={"name": "name"})
+        self.assertEqual(response.status_code, 401)
+
+    def test_post_user_with_taken_email(self):
+        token = self.login().json['token']
+        response = self.client.post('/api/users',
+                                    headers={'token': token},
+                                    data={"email": "email1"})
+        self.assertEqual(response.status_code, 400)
+
+    def test_valid_post_user(self):
+        new_user = {
+            "name": "name1",
+            "password": "pass1",
+            "email": "email321",
+            "is_admin": "0"
+        }
+        token = self.login().json['token']
+        response = self.client.post('/api/users',
+                                    headers={'token': token},
+                                    data=new_user)
+        self.assertEqual(response.status_code, 201)
+        new_user["password"] = "pass1"
```

Comment:
also you need to check is data in db the same as you sent, especially pay attention for boolean field